### DISCUSSION TITLE
Keyring method "wallet_invokeKeyring" does not exist

### DISCUSSION THREAD

**Discussion by leopashov** [MetaMask/snaps#1887] [General]
Tried cloning and running the simple-keyring repo at https://github.com/MetaMask/snap-simple-keyring/tree/main.
Cloned repo, ran './scripts/cleanup.sh' followed by 'yarn install' and 'yarn start'.
Opened localhost:8000 in microsoft edge, and connected metamask flask (V 11.2.0-flask.0). Clicked 'create account' and got: 

'{
  "code": -32601,
  "message": "The method \"wallet_invokeKeyring\" does not exist / is not available.",
  "data": {
    "origin": "http://localhost:8000"
  },
  "stack": "{\n  \"code\": -32601,\n  \"message\": \"The method \\\"wallet_invokeKeyring\\\" does not exist / is not available.\",\n  \"data\": {\n    \"origin\": \"http://localhost:8000\"\n  },\n  \"stack\": \"Error: The method \\\"wallet_invokeKeyring\\\" does not exist / is not available.\\n    at new i (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-3.js:16:10940)\\n    at s (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-3.js:16:13880)\\n    at Object.methodNotFound (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-3.js:16:14441)\\n    at n.methodNotFound (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-0.js:1:351456)\\n    at v.getTypedPermissionSpecification (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-0.js:1:335931)\\n    at v.getRestrictedMethod (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-0.js:1:336172)\\n    at v.<anonymous> (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-0.js:1:357406)\\n    at Generator.next (<anonymous>)\\n    at chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-0.js:1:356881\\n    at new Promise (<anonymous>)\\n    at s (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-0.js:1:356626)\\n    at chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-0.js:1:357318\\n    at chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-4.js:19:37941\\n    at chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-4.js:19:36950\\n    at new Promise (<anonymous>)\\n    at a._runMiddleware (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-4.js:19:36607)\\n    at a._runAllMiddleware (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-4.js:19:36513)\\n    at async a._processRequest (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-4.js:19:36294)\\n    at async a._handle (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-4.js:19:36135)\"\n}\n  at new i (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-3.js:16:10940)\n  at s (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-3.js:16:13880)\n  at Object.methodNotFound (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-3.js:16:14441)\n  at n.methodNotFound (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-0.js:1:351456)\n  at v.getTypedPermissionSpecification (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-0.js:1:335931)\n  at v.getRestrictedMethod (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-0.js:1:336172)\n  at v.<anonymous> (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-0.js:1:357406)\n  at Generator.next (<anonymous>)\n  at chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-0.js:1:356881\n  at new Promise (<anonymous>)\n  at s (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-0.js:1:356626)\n  at chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-0.js:1:357318\n  at chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-4.js:19:37941\n  at chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-4.js:19:36950\n  at new Promise (<anonymous>)\n  at a._runMiddleware (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-4.js:19:36607)\n  at a._runAllMiddleware (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-4.js:19:36513)\n  at async a._processRequest (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-4.js:19:36294)\n  at async a._handle (chrome-extension://ljfoeinjpaedjfecbmggjgodbgkmjkjk/common-4.js:19:36135)"
}'

Any guidance on how to deal with this would be greatly appreciated!

**Comment by montelaidev** [comment]
Hi @leopashov, the latest version of the `snap-simple-keyring` requires flask version `11.4.0` or above. Its currently not published onto the chrome store yet. You can try to build the extension locally right now by cloning the repo, and running `yarn start:flask`, or try to install this [build](https://github.com/MetaMask/metamask-extension/pull/21364#issuecomment-1761753445). 

Thanks for raising this, we will add a extension version requirement onto the `snap-simple-keyring`

**Reply by leopashov** [reply]
Thank you very much. I managed to install that build but now get: 
'{
  "code": -32603,
  "message": "Snap \"local:http://localhost:8080\" is not permitted to handle requests from \"http://localhost:8000\".",
  "data": {
    "originalError": {}
  }
}' 

Is there a workaround/ solution you could provide for this?
Many thanks :)

**Reply by Montoya** [reply]
Have you installed the Snap? The dapp needs to request a connection to the Snap and the Snap needs to be installed into your MetaMask Flask instance. https://docs.metamask.io/snaps/reference/rpc-api/#wallet_requestsnaps

**Reply by leopashov** [reply]
Snap seems to install fine: 
![image](https://github.com/MetaMask/snaps/assets/58823129/69c05e19-0783-4380-80a9-ba051d756642)

![image](https://github.com/MetaMask/snaps/assets/58823129/d4b9e727-b829-49de-ba65-1674124ac32e)

But then clicking on any of the methods, e.g 'Create Account' give the error.

**Reply by AlexJupiter** [reply]
@leopashov please make sure you're using MetaMask Flask and not MetaMask stable for development. Perhaps that will solve some of your problems.

**Reply by leopashov** [reply]
Have been using flask throughout, will continue to keep digging
